src/Module.ts

```diff
@@ -716,6 +716,27 @@ export default class Module {
 		this.includeAllExports(false);
 	}
 
+	includeExportsByNames(names: readonly string[], includeNamespaceMembers: boolean): void {
+		if (!this.isExecuted) {
+			markModuleAndImpureDependenciesAsExecuted(this);
+			this.graph.needsTreeshakingPass = true;
+		}
+
+		for (const name of names) {
+			const variable = this.getVariableForExportName(name)[0];
+			if (variable) {
```

Comment:
The case where `variable` is `null` corresponds to the situation where we destructure something that is not actually exported. This is fine—we just get `undefined`—but I think this should be covered in a test. I.e.

```js
const {doesNotExist} = import('./foo.js');
```